doc/hardware/peripherals/mspi.rst

```diff
@@ -0,0 +1,198 @@
+which user callback to trigger based on :c:enum:`mspi_bus_event_cb_mask` upon completion
+of each async/sync transfer if the callback had been registered using
+:c:func:`mspi_register_callback`. Or not to trigger any callback at all with
+:c:enum:`MSPI_BUS_NO_CB` even if the callbacks are already registered.
+In which case that a controller supports hardware command queue, user could take the full
+advantage of it in terms of performance if scatter IO and callback management are supported.
+
+Device Tree
+===========
+
+Here is an example for defining an MSPI controller in device tree:
+The mspi controller's bindings should reference mspi-controller.yaml as one of the base.
+
+.. code-block:: devicetree
+
+   mspi0: mspi@400 {
+            status = "okay";
+            compatible = "zephyr,mspi-emul-controller";
+
+            reg = < 0x400 0x4 >;
+            #address-cells = < 0x1 >;
+            #size-cells = < 0x0 >;
+
+            clock-frequency = < 0x17d7840 >;
+            op-mode = "MSPI_MASTER";
+            duplex = "MSPI_HALF_DUPLEX";
+            ce-gpios = < &gpio0 0x5 0x1 >, < &gpio0 0x12 0x1 >;
+            dqs-support;
+
+            pinctrl-0 = < &pinmux-mspi0 >;
+            pinctrl-names = "default";
+   };
+
+Here is an example for defining an MSPI device in device tree:
+The mspi device's bindings should reference mspi-device.yaml as one of the base.
+
+.. code-block:: devicetree
+
+   &mspi0 {
+
+            mspi_dev0: mspi_dev0@0 {
+                     status = "okay";
+                     compatible = "zephyr,mspi-emul-device";
+
+                     reg = < 0x0 >;
+                     size = < 0x10000 >;
+
+                     mspi-max-frequency = < 0x2dc6c00 >;
+                     mspi-io-mode = "MSPI_IO_MODE_QUAD";
+                     mspi-data-rate = "MSPI_SINGLE_DATA_RATE";
+                     mspi-hardware-ce-num = < 0x0 >;
+                     read-instruction = < 0xb >;
+                     write-instruction = < 0x2 >;
+                     instruction-length = "INSTR_1_BYTE";
+                     address-length = "ADDR_4_BYTE";
+                     rx-dummy = < 0x8 >;
+                     tx-dummy = < 0x0 >;
+                     xip-config = < 0x0 0x0 0x0 0x0 >;
+                     ce-break-config = < 0x0 0x0 >;
+            };
+
+   };
+
+User should specify target operating parameters in the DTS such as mspi-max-frequency,
+mspi-io-mode and mspi-data-rate even though they may subject to change during run-time.
+Typically it should represent the condition of the device during normal operations.
+
+Muti-slave
+==========
+With :c:struct:`mspi_dev_id` defined as collection of the device index and CE GPIO from
+device tree, the API supports multiple devices on the same controller instance.
+It is up to the controller driver implementation whether to support device switching
+either by software or by hardware. If by software, the switching should be performed
+in call to :c:struct:`mspi_dev_cfg`.
```

Comment:
The controller driver implementation may or may not support device switching,
which can be performed either by software or by hardware. If the switching is handled
by software, it should be performed in a call to :c:struct:`mspi_dev_cfg`.